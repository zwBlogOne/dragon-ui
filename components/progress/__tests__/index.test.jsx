import React from 'react';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Progress from '../index';

describe('Progress', () => {
  it('renders normal Progress correctly', () => {
    const wrapper = render(
      <div>
        <Progress percent="30" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders different themes of Progress correctly', () => {
    const wrapper = render(
      <div>
        <Progress percent="30" />
        <Progress percent="30" theme="info" />
        <Progress percent="30" theme="success" />
        <Progress percent="30" theme="warning" />
        <Progress percent="30" theme="error" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders Progress with radius & round props correctly', () => {
    const wrapper = render(
      <div>
        <Progress percent="30" size="lg" theme="info" />
        <Progress percent="30" radius size="lg" theme="info" />
        <Progress percent="30" round size="lg" theme="info" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('renders different sizes of Progress correctly', () => {
    const wrapper = render(
      <div>
        <Progress percent="30" size="xl" theme="info" />
        <Progress percent="30" size="lg" theme="info" />
        <Progress percent="30" theme="info" />
        <Progress percent="30" size="sm" theme="info" />
        <Progress percent="30" size="xs" theme="info" />
      </div>
    );

    expect(toJson(wrapper)).toMatchSnapshot();
  });
});

