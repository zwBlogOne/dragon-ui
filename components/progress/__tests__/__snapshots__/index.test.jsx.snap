// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress renders Progress with radius & round props correctly 1`] = `
<div>
  <div
    class="ui-progress theme-info size-lg"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress radius theme-info size-lg"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress round theme-info size-lg"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
</div>
`;

exports[`Progress renders different sizes of Progress correctly 1`] = `
<div>
  <div
    class="ui-progress theme-info size-xl"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-info size-lg"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-info"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-info size-sm"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-info size-xs"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
</div>
`;

exports[`Progress renders different themes of Progress correctly 1`] = `
<div>
  <div
    class="ui-progress theme-default"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-info"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-success"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-warning"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
  <div
    class="ui-progress theme-error"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
</div>
`;

exports[`Progress renders normal Progress correctly 1`] = `
<div>
  <div
    class="ui-progress theme-default"
  >
    <div
      class="ui-progress-line-outer"
    >
      <div
        class="ui-progress-line-inner"
      >
        <div
          class="ui-progress-bg"
          style="width:30%"
        />
      </div>
    </div>
    <span
      class="ui-progress-line-text"
    >
      30%
    </span>
  </div>
</div>
`;
